ndm-exporter/collector/seachest.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"github.com/openebs/node-disk-manager/db/kubernetes"
+	seachestmetrics "github.com/openebs/node-disk-manager/pkg/metrics/seachest"
+	"github.com/openebs/node-disk-manager/pkg/seachest"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+// SeachestMetricCollector contains the metrics, concurrency handler and client to get the
+// metrics from seachest
+type SeachestMetricCollector struct {
```

Comment:
Should we keep this generic. For example this is SMART metrics that we are getting and in the current implementation we are using SeaChest to get the values.

Reply:
Its more towards the library we use. We can use seachest library for some metrics. SMART for some other metrics. There will be some metrics that overlaps. But I think a library specific metric will be good, even if they overlaps. This will give the option to the user on which metric he need to choose.

If we think in angle of node exporter, there will be some disk related metrics that will be exposed by both node-exporter and NDM. But the user can choose the metric using his query, rather than we limiting the user to metrics from a certain lib only.